crates/revm/src/db/in_memory_db.rs

```diff
@@ -86,26 +100,25 @@ impl<ExtDB: DatabaseRef> CacheDB<ExtDB> {
 // TODO It is currently only committing to cached in-memory DB
 impl<ExtDB: DatabaseRef> DatabaseCommit for CacheDB<ExtDB> {
     fn commit(&mut self, changes: Map<H160, Account>) {
+        // clear storage by setting all values to `0`
+        fn clear_storage(storage: &mut HashMap<U256, U256>) {
+            storage.values_mut().for_each(|val| {
+                *val = U256::zero();
+            })
+        }
+
         for (add, acc) in changes {
             if acc.is_empty() || matches!(acc.filth, Filth::Destroyed) {
-                self.cache.remove(&add);
-                self.storage.remove(&add);
+                // clear account data, but don't remove entry
+                self.changes.insert(add, Default::default());
+                self.storage.get_mut(&add).map(clear_storage);
             } else {
-                self.insert_cache(add, acc.info);
+                self.insert_change(add, acc.info);
```

Comment:
same q here, what's the diff between cache / change?

Reply:
It is because of this: https://github.com/bluealloy/revm/pull/126#issuecomment-1152717104
the `cache` is loaded from the external DB.
the `changes` is what is applied from `commit`.